src/VirtoCommerce.NotificationsModule.Data/TemplateLoaders/FileSystemTemplateLoaderOptions.cs

```diff
@@ -0,0 +1,9 @@
+using System.Text.RegularExpressions;
+
+namespace VirtoCommerce.NotificationsModule.Data.TemplateLoaders
+{
+    public class FileSystemTemplateLoaderOptions
+    {
+        public Regex TemplateFilePattern { get; set; } = new Regex(@"(?<type>[\w-]+)_(?<part>body|subject)(?<lang>\.[a-z]{2}(-[A-Z]{2})?)?\.[\w]+$", RegexOptions.IgnoreCase);
```

Comment:
Incorrect pattern for `lang`.
Both language and country codes can be more than 2 characters (`quz-BO`, `en-029`).
Also, there can be more than 2 parts (`sr-Cyrl-BA`).

You should build the template file name when you know the notification type and the culture code instead of parsing a template file name.